prover/src/lib.rs

```diff
@@ -182,6 +182,7 @@ pub trait Prover {
         trace_info: &TraceInfo,
         main_trace: &ColMatrix<Self::BaseField>,
         domain: &StarkDomain<Self::BaseField>,
+        num_partitions: usize,
     ) -> (Self::TraceLde<E>, TracePolyTable<E>)
     where
         E: FieldElement<BaseField = Self::BaseField>;
```

Comment:
I don't think we need to pass `num_partitions` here as we can get this info from `self.options()`.

Reply:
Yes, indeed, but it just feels more congruent with how it is done for the auxiliary trace and the constraint trace.

Reply:
In `set_aux_trace()` we use `partition_size` no? It seems to me that here (and also in the examples), the variable is named `num_partitions` while the value passed in is `partition_size`.

I think having `PartitionOptions` structure would help avoid such confusions and so would be a bit cleaner.